core/rawdb/accessors_chain.go

```diff
@@ -29,7 +29,14 @@ import (
 
 // ReadCanonicalHash retrieves the hash assigned to a canonical block number.
 func ReadCanonicalHash(db common.Database, number uint64) common.Hash {
-	data, _ := db.HeaderTable().Get(numKey(number))
+	var data []byte
+	Must("get canonical hash", func() (err error) {
+		data, err = db.HeaderTable().Get(numKey(number))
+		if err == common.ErrNotFound {
```

Comment:
Many cases were ignoring errors and considering empty data to mean 'not found' - but we can't ignore all the errors since they could be retryable, so we check for `common.ErrNotFound`.